src/parser.c

```diff
@@ -1279,38 +1279,26 @@ enum parse_rv parseTransaction(struct TransactionState *const state, parser_meta
             meta->type_id = convert_type_id_to_type(meta->raw_type_id, meta->chain);
             state->state++;
 
-            switch (meta->chain) {
-            case CHAIN_X:
-              if (!is_xchain_transaction(meta->type_id.x)) {
-                REJECT("Blockchain ID did not match expected value for network ID");
-              }
-              break;
-            case CHAIN_P:
-              if (!is_pchain_transaction(meta->type_id.p)) {
-                REJECT("Blockchain ID did not match expected value for network ID");
-              }
-              break;
-            case CHAIN_C:
-              if (is_pchain_transaction(meta->type_id.p)) {
-                REJECT("Transaction ID indicates P-chain but blockchain ID is is not 0");
-              } else if (!is_cchain_transaction(meta->type_id.c)) {
-                REJECT("Blockchain ID did not match expected value for network ID");
-              }
-              break;
-            }
-
             INIT_SUBPARSER(baseTxState, BaseTransaction);
             label_t label = type_id_to_label(meta->type_id, meta->chain);
             if (ADD_PROMPT("Sign", label.label, label.label_size, strcpy_prompt)) break;
         } fallthrough;
         case 3: { // Base transaction
-            if(meta->chain != CHAIN_C) { // C-chain atomic transactions have a different format; skip here.
-                PRINTF("TRACE\n");
+            switch (meta->chain) {
+              case CHAIN_X:
+              case CHAIN_P:
+                PRINTF("TRACE pre basic tx subparser break, chain enum: %d\n", meta->chain);
                 CALL_SUBPARSER_BREAK(baseTxState, BaseTransaction);
```

Comment:
This has `break`, so we do the "bubble switch break" below now that there is new switch.